cuda/scatter_kernel.cu

```diff
@@ -92,8 +92,11 @@ __global__ void arg_kernel(at::cuda::detail::TensorInfo<scalar_t, int64_t> src,
     IndexToScatterOffsets4<scalar_t, scalar_t, int64_t, Dims>::compute(
         i, dim, index, &indexOffset, src, &srcOffset, out, &outOffset, arg,
         &argOffset);
+    __syncthreads();
```

Comment:
Do we really need this call here? We also parallelize in the block-level anyway.

Reply:
I think, I forget to remove this line. Using the atomMax you provided, we can just avoid the problems, when several values are all the maximum the argmax index would be random selected

Reply:
Yes, you are right. This is indeed needed.